### ISSUE TITLE
feature request: structured log

### ISSUE THREAD

**Issue by Nymphium** [mirage/ocaml-cohttp#872]
Now cohttp debug log is so unstructured that it is hard to see in for example datadog.
I think it is preferred to use `Logs.Tag` for several information, url, http method, status, and etc.

**Comment by mseri** [comment]
Refactoring the code to enable proper logging is high on the list of welcome features. If anybody has the time to put in the work or start a discussion on the design, it would be very welcome

**Comment by rgrinberg** [comment]
Indeed. In particular, we'd prefer to kick out logging from the core of the system. It should be possible to add your own logging in a modular way at the transport layer. We should only be logging protocol level changes.

**Comment by Nymphium** [comment]
>  It should be possible to add your own logging in a modular way at the transport layer. 

Yes, I'm for the statement too, but the both should be done. Even if the logger is inserted moduraly, the information passed is unstructured, the log is still unstructured.
